lib/MySQL_Session.cpp

```diff
@@ -185,6 +255,40 @@ void MySQL_Session::writeout() {
 	proxy_debug(PROXY_DEBUG_NET,1,"Thread=%p, Session=%p -- Writeout Session %p\n" , this->thread, this, this);
 }
 
+bool MySQL_Session::handler_special_queries(PtrSize_t *pkt) {
+	if (pkt->size==SELECT_VERSION_COMMENT_LEN+5 && strncmp((char *)SELECT_VERSION_COMMENT,(char *)pkt->ptr+5,pkt->size-5)==0) {
+		PtrSize_t pkt_2;
+		pkt_2.size=PROXYSQL_VERSION_COMMENT_LEN;
+		pkt_2.ptr=l_alloc(pkt_2.size);
+		memcpy(pkt_2.ptr,PROXYSQL_VERSION_COMMENT,pkt_2.size);
+		status=WAITING_CLIENT_DATA;
+		client_myds->DSS=STATE_SLEEP;
+		client_myds->PSarrayOUT->add(pkt_2.ptr,pkt_2.size);
```

Comment:
It is indeed a hardcoded packet. This is to handle special queries that shouldn't need a backend to run.